content/blog/pulumi-kubernetes-operator/index.md

```diff
@@ -0,0 +1,529 @@
+            }
+        });
+    }
+}
+```
+
+{{% /choosable %}}
+
+{{% choosable language go %}}
+
+```go
+Coming Soon
+```
+
+{{% /choosable %}}
+{{% /chooser %}}
+
+## CI/CD Scenarios
+
+As you build out your Kubernetes workloads, you'll need to implement update strategies
+to ship new software easily and reliably. Kubernetes API resources like
+[Deployments][k8s-deployment] are used to model stateless apps and operate their lifecycles
+e.g. recreate app replicas, or perform a rolling update.
+
+A common approach that most teams rely on is the [Blue / Green][blue-green] deployment.
+This starts with an initial version of the app, and then deploys an updated version
+alongside the initial version. Once ready, traffic is switched over from the
+initial version to the new version with no downtime.
+
+![Blue Green Kubernetes Deployment](./blue-green.png)
+
+In Kubernetes, Blue/Green is modeled by using a [Service][k8s-service]
+that load balances and selects a set of Deployment pods labeled "blue," and then switching to a
+different selection of updated, "green" pods.
+
+Similar to the Stack CustomResources in the previous sections, we can deploy a Stack
+of a [Blue/Green Kubernetes app in Pulumi][blue-green-example], and step through a sequence of its Git commits
+similiar to how a CI/CD pipeline does. Additionally, we can use the [Pulumi config system][pulumi-config] to parameterize
+settings and secrets on how to manage the Stack CustomResource project.
+
+Deploying the active blue and passive green versions of the app can be done on
+the example Stack with the following steps.
+
+```bash
+pulumi config set --secret pulumiAccessToken <YOUR_PULUMI_API_TOKEN>
+pulumi config set stackProjectRepo https://github.com/metral/pulumi-blue-green
+pulumi config set stackCommit b19759220f25476605620fdfffeface39a630246
+pulumi up -y
+```
+
+After deployment, we'll test that the v1 / "blue" endpoint is live and running.
+The v2 / "green" app will be on standby.
+
+```bash
+while true; do curl http://34.83.25.150:80/version ; echo; sleep 1; done
+
+{"id":1,"content":"current"}
+{"id":1,"content":"current"}
+{"id":1,"content":"current"}
+{"id":1,"content":"current"}
+...
+```
+
+When we're ready to switch traffic, the blue/green
+transition is done by updating the config to the Git new commit of the selector
+swap, and running an update.
+
+```bash
+pulumi config set stackCommit f4bf2f7b54ec441d5af374933cca4ef09f2bce24
+pulumi up -y
+```
+
+After a few seconds you should start seeing the generated traffic returning the
+new version in a clean cut-over with no dropped packets.
+
+```bash
+...
+{"id":1,"content":"current"}
+{"id":1,"content":"current"}
+{"id":1,"content":"current"}
+{"id":2,"content":"new"}
+{"id":2,"content":"new"}
+{"id":2,"content":"new"}
+{"id":2,"content":"new"}
+...
+```
+
+Check out the [full walkthrough][blue-green-walkthrough] for a tutorial, and
+the video clip below for a demo.
+
+TODO: add video-clip blue/green stack demo.
```

Comment:
Is there a link for this now?

Reply:
The video is in [Gdrive](https://drive.google.com/drive/u/1/folders/156HJjYlQudFOdrPRrVHokJhPlZnoiRFU). Will be uploaded to YT soon